src/sentry/static/sentry/app/views/organizationIntegrations/sentryAppInstallations.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {groupBy} from 'lodash';
+import parseurl from 'parseurl';
+import qs from 'query-string';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import AsyncView from 'app/views/asyncView';
+import {Client} from 'app/api';
+import SentryApplicationRow from 'app/views/settings/organizationDeveloperSettings/sentryApplicationRow';
+import {t} from 'app/locale';
+
+const api = new Client();
+
+export default class SentryAppInstallations extends AsyncView {
+  getEndpoints() {
+    let {orgId} = this.props;
+
+    return [
+      ['applications', `/organizations/${orgId}/sentry-apps/`],
+      ['installs', `/organizations/${orgId}/sentry-app-installations/`],
+    ];
+  }
+
+  redirectUser = data => {
+    const {install, app} = data;
+    const {installs} = this.state;
+
+    if (!app.redirectUrl) {
+      addSuccessMessage(t(`${app.slug} successfully installed.`));
+      this.setState({installs: [install, ...installs]});
+    } else {
+      const url = parseurl({url: app.redirectUrl});
+      // Order the query params alphabetically.
+      // Otherwise ``qs`` orders them randomly and it's impossible to test.
+      const installQuery = JSON.parse(
+        JSON.stringify({installationId: install.uuid, code: install.code})
+      );
+      // const query = Object.assign(qs.parse(url.query), installQuery);
+      const query = {...qs.parse(url.query), ...installQuery};
+      const redirectUrl = `${url.protocol}//${url.host}${url.pathname}?${qs.stringify(
+        query
+      )}`;
+      window.location.assign(redirectUrl);
+    }
+  };
+
+  install = app => {
+    const {orgId} = this.props;
+
+    const success = install => {
+      this.redirectUser({install: {...install}, app: {...app}});
+    };
+
+    const error = err => {
+      addErrorMessage(t(`Unable to install ${app.name}`));
+    };
+
+    const opts = {
+      method: 'POST',
+      data: {slug: app.slug},
+      success,
+      error,
+    };
+
+    api.request(`/organizations/${orgId}/sentry-app-installations/`, opts);
```

Comment:
Would be great to have actionCreator functions for the API calls. That would let the component focus on rendering and let the creator handle api calls and indicator messages.

Reply:
Going to add to https://github.com/getsentry/sentry/pull/10933